contracts/token/ERC6909/extensions/draft-ERC6909ContentURI.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {ERC6909} from "../draft-ERC6909.sol";
+import {IERC6909ContentURI} from "../../../interfaces/draft-IERC6909.sol";
+
+/**
+ * @dev Implementation of the Content URI extension defined in ERC6909.
+ */
+contract ERC6909ContentURI is ERC6909, IERC6909ContentURI {
+    string private _contractURI;
+    mapping(uint256 id => string) private _tokenURIs;
+
+    /// @inheritdoc IERC6909ContentURI
+    function contractURI() external view virtual override returns (string memory) {
+        return _contractURI;
+    }
+
+    /// @inheritdoc IERC6909ContentURI
+    function tokenURI(uint256 id) external view virtual override returns (string memory) {
+        return _tokenURIs[id];
+    }
+
+    function _setContractURI(string memory newContractURI) internal {
+        _contractURI = newContractURI;
+    }
+
+    function _setTokenURI(uint256 id, string memory newTokenURI) internal {
+        _tokenURIs[id] = newTokenURI;
+    }
```

Comment:
Should we emit an event on token URI set?

https://ethereum-magicians.org/t/erc-6909-multi-token-standard/13891/2?u=arr00

Reply:
Yes, we should emit an event for state changes. I would've proposed `ERC6909MetadataUpdated` but I just noticed there's [ERC4906](https://eips.ethereum.org/EIPS/eip-4906) which standardized `MetadataUpdate` for ERC-721. I don't see much issue here considering that it also defines `BatchMetadataUpdate`.

So yeah, I would agree with `MetadataUpdate`